File: algorithms/001_two_sum_2.go
package main

import "fmt"

func main() {
	fmt.Println(twoSum([]int{2, 7, 11, 15}, 9))
}

func twoSum(nums []int, target int) []int {
	numsMap := make(map[int]int)

	for index, value := range nums {
		numDiff := target - value
		if v, ok := numsMap[numDiff]; ok {
			return []int{v, index}
		}
		numsMap[value] = index
	}

	return []int{}
}

File: algorithms/001_two_sum_2.php
<?php
declare(strict_types=1);

/**
 * @param int[] $nums
 * @param int $target
 * @return int[]
 */
function twoSum(array $nums, int $target): array
{
    $numsMap = [];

    foreach ($nums as $key => $num) {
        $numDiff = $target - $num;
        if (array_key_exists($numDiff, $numsMap)) {
            return [$numsMap[$numDiff], $key];
        }
        $numsMap[$num] = $key;
    }

    return [];
}

var_dump(twoSum([2, 7, 11, 15], 9));
